### ISSUE TITLE
about training fairness

### ISSUE THREAD

**Issue by yzf-code** [LirongWu/MAPE-PPI#11]
https://github.com/LirongWu/MAPE-PPI/blob/c11708b4b36ade89448606a4a5a42e6e6e25d59a/src/dataloader.py#L98C5-L98C51
Dear authors, I have a question about the division of the dataset.
In the code, I saw that the PPI graph that the GIN model relies on is constructed from the entire dataset, including the nodes and edges of the validation set and the test set. During the training process, only the PPI edges of the train split are used to calculate the loss. Is this true?
If so, does it mean that the feature calculation of the nodes in the train split actually depends on the points from the validation set and the test set? Will this bring about fairness issues?
Looking forward to your reply, thank you very much!

**Comment by yzf-code** [comment]
I noticed in GNN-PPI, they mentioned different PPI Network Graph Construction in section 4.3 
"There are two graph construction methods, graph construct by all data (GCA, G = (P, X )) and graph construct by trainset (GCT,G = (Pv , X_train)). " 
It seems GCT is closer to the discovery of novel protein interactions.

**Comment by LirongWu** [comment]
The issue you mention is actually of two settings in graph machine learning: (1) transductive and (2) inductive, and they differ in terms of validate and test features are usable in the training process. The setup in this paper follows HIGH-PPI and is closer to the former (1) transductive. if it is a strictly protein discovery rather than classification problem, probably the latter is more appropriate. Table 1 of the GNN-PPI paper also uses the GCA setting.

**Comment by yzf-code** [comment]
Thanks! I got it. But I have two more questions.
1. It seems most of the GNN-based PPI prediction methods use transductive setting?  e.g. GNN-PPI, HIGH-PPI, MAPE-PPI.
2. Besides, How did you obtain the experimental results of SemiGNN-PPI? It seems that they did not publish the code, and the results in your table 1 are inconsistent with what they reported.
thank you very much!

**Comment by LirongWu** [comment]
1. Yes, the GNN-based model is more advantageous in transductive setting.
2. They don't make their code publicly available (we sent a lot of emails without getting any reply). Thus, we spent a lot of time reproducing their results, so the results are not exactly consistent because we can't know their data splits.

**Comment by yzf-code** [comment]
haha, I got it. Thanks for your reply!
